.github/workflows/release-build-native-linux.yml

```diff
@@ -0,0 +1,70 @@
+name: release-build-native-linux
+
+on:
+  release:
+    types: [created] 
+  push:
+    branches:
+      - master
+  pull_request:
+    branches:
+      - master
+
+jobs:
+# Trigger this job only on Release
+  build-on-release:
+    if: github.event_name=='release'&& github.event_name != 'pull_request'&& github.event_name != 'push'
+    name: Build on Release
+    runs-on: ubuntu-latest 
+
+    steps:
+    - uses: actions/checkout@v2
+    - name: Install graalvm
+      uses: DeLaGuardo/setup-graalvm@3
+      with:
+        graalvm-version: '20.0.0.java11'
+    - name: Install native-image
+      run: gu install native-image
+    - name: Set version
+      run: ./mvnw versions:set -DnewVersion="${{ github.event.release.tag_name }}" 
+    - name: Build native executable
+      run: ./mvnw package -Dnative 
+    - name: Upload native executable
+      id: upload-native-executable
+      uses: actions/upload-release-asset@v1
+      env:
+        GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+      with:
+        upload_url: ${{ github.event.release.upload_url }}
+        asset_path: ./target/starfish-${{ github.event.release.tag_name }}-runner
+        asset_name: starfish-linux
+        asset_content_type: application/octet-stream
+
+# Trigger this job only on Push or Pull-Request
+  build-on-push-pr:
+    if: github.event_name!='release'&& (github.event_name == 'pull_request'|| github.event_name == 'push')
+    name: Build on Push or PR
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - name: Install graalvm
+        uses: DeLaGuardo/setup-graalvm@3
+        with:
+          graalvm-version: '20.0.0.java11'
+      - name: Install native-image
+        run: gu install native-image
+      - name: Set version
+        run: ./mvnw versions:set -DnewVersion="1.0.0" 
+      - name: Build native executable
+        run: ./mvnw package -Dnative 
+      - name: Upload native executable
+        id: upload-native-executable
+        uses: actions/upload-artifact@v2
+        with:
+          name: starfish-linux
+          path: ./target/starfish-1.0.0-runner
```

Comment:
use the PR number in the version number here ?

Reply:
use same version as used in the build.

otherwise you'll end up with duplicate binaries...